front/components/agent_builder/instructions/InstructionsTipsPopover.tsx

```diff
@@ -0,0 +1,152 @@
+import {
+  ContentMessage,
+  PopoverContent,
+  PopoverRoot,
+  PopoverTrigger,
+  Spinner,
+} from "@dust-tt/sparkle";
+import { LightbulbIcon } from "@dust-tt/sparkle";
+import { Button } from "@dust-tt/sparkle";
+import { useEffect, useState } from "react";
+import React from "react";
+
+import type {
+  APIError,
+  BuilderSuggestionsType,
+  Result,
+  WorkspaceType,
+} from "@app/types";
+import { Err, Ok } from "@app/types";
+
+const STATIC_TIPS = [
+  "Break down your instructions into steps to leverage the model's reasoning capabilities.",
+  "Give context on how you'd like the agent to act, e.g. 'Act like a senior analyst'.",
+  "Add instructions on the format of the answer: tone of voice, answer in bullet points, in code blocks, etc...",
+];
+
+type TipStatus = "loading" | "loaded" | "error";
+
+interface InstructionTipsPopoverProps {
+  owner: WorkspaceType;
+  instructions: string;
+}
+
+export function InstructionTipsPopover({
+  owner,
+  instructions,
+}: InstructionTipsPopoverProps) {
+  const [tips, setTips] = useState<string[]>(STATIC_TIPS);
+  const [status, setStatus] = useState<TipStatus>("loaded");
+  const [error, setError] = useState<APIError | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
```

Comment:
Instead of useEffect hook, can we run this on onOpenChange? We have been using it wrongly and I want to be sure we do it right for agent builder 🥲